### ISSUE TITLE
More compact UI

### ISSUE THREAD

**Issue by ffMathy** [ffMathy/Shapeshifter#307]
I thought about something. Currently, the only reason clipboard items are so big visually is because the text data type item _sometimes_ has more than 1 line.

But how often is this? Often I find myself copying smaller regions of text, and even so, I only look at a small subset of text to identify it.

In #245 we will also have a diffing mechanism.

With that, I propose to change the maximum lines to 1, and then make each clipboard item more compact. This will result in more items in the list also.

**Comment by thegabriestein** [comment]
I think this is a similar issue as #54, that seeing preview of the content you're going to paste is valuable, but you don't necessarily want to see it full size all the time. Scrolling magnification would help with this as you scroll over it. Also, most items are larger because of the icons on the left (like the folder icons etc). 

These help you visually identify them, maybe they could just float instead of span the whole area? And it might be nice to let items take up more or less vertical space as needed, but there is a strain on rapid assessment when a list is filled with uneven items.

I just checked and out of a screens worth of items (about 19), 7 of them were 2+ lines, so that's a decent amount for me, although I'm working on some notes and courses though so that may skew it. 

Hence maybe just showing a single line and then showing more if you select it or use an action menu item of "Preview"?

Simple solution for now though:
Item default takes up 1 content line, and can expand up to 3 if it needs it. 

This should be the best of both worlds, saving space for most items, and allowing larger content to take up a little more space if needed.

**Comment by ffMathy** [comment]
Hmmm, maybe experimenting with this scroll magnification is worth it. But I generally don't like a jumpy UI, and it will be jumpy if each element resizes all the time.

**Comment by thegabriestein** [comment]
Well if:
1. Only one item expands (instead of all of them swelling or shrinking)
2. That one item is always in an expanded state, and when you scroll or hover on an item, it switches to that being the expanded item

Then:
It should prevent it being too jumpy. 

I'm not super sold on this though, definitely discussion stage. Assuming would take some work to implement, so don't want to prioritize asap. It seems like it might help with previewing items in a very streamlined way without any extra keybinds or steps and only if needed. And if the top item was always auto-expanded, then the most recent and commonly repasted item would be expanded like this:

![](https://i.imgur.com/u2glTCg.png)

Then you could just scroll down and the same vertical and horizontal space would be taken up so it wouldn't have a "jumpy" movement effect (which would be really annoying). 

Also just noticed #341 regarding the double pasting and single preview lines.

Ooh, what about this:
You press Left Arrow which expands the item, and the action menu appears from below the item? Then Right Arrow hides the action menu and shrinks the preview area again?

![](https://i.imgur.com/GTyCF9y.png)

**Comment by ffMathy** [comment]
I don't think it intuitively makes sense to pop up the list below the current item when pressing the left key.

As for the jumpiness, the thing I am concerned about is not the current item (it wouldn't jump as you say), but the items below it. Those are also important when needing to know where to go in the list.
